lib/command.js

```diff
@@ -887,6 +896,28 @@ Expecting one of '${allowedValues.join("', '")}'`);
     return userArgs;
   }
 
+  /**
+   * @param {boolean} async
+   * @param {Function} userArgsCallback
+   * @param {string[]} [argv]
+   * @param {Object} [parseOptions]
+   * @param {string} [parseOptions.from]
+   * @return {Command|Promise}
+   * @api private
+   */
+
+  _parseSubroutine(async, userArgsCallback, argv, parseOptions) {
+    this._asyncParsing = async;
+    const methodName = async ? 'parseAsync' : 'parse';
+    if (!PRODUCTION && this.parent) {
+      console.warn(`Called .${methodName}() on subcommand '${this._name}'.
```

Comment:
I don't think I care enough about this error to add. For interest, did you hit this in real world usage?

If I did add it, I think I would rather have some small code repetition than an extra level of code.

Reply:
> I don't think I care enough about this error to add.

It seems a little weird to add the error in `addCommand()` but not this one. But everything could be fixed by implementing the suggestion from [the first comment](https://github.com/tj/commander.js/pull/1917#issuecomment-1656365692). Why do you dislike it? I think it is very easy to implement. If I'm not mistaken, it would require even fewer lines of code than the combination of this warning and the error in `addCommand()` (cannot check it right now unfortunately since I'm on a small trip and haven't taken my laptop with me). Only the error from `passThroughOptions()` would need to be turned into a warning in `_parseSubroutibe()`, and something done about the `parent` use in `outputHelp()`.

> For interest, did you hit this in real world usage?

No, just trying on the library developer mindset and thinking of different ways it could be used 🙂

> If I did add it, I think I would rather have some small code repetition than an extra level of code.

More shared code can be added to `_parseSubroutibe()` later, like after the suggestion from the first comment is implemented or #1919 is merged.

Reply:
The high level author error that does sometimes happen is chucking `parse()` on the end of the subcommand definition. This would catch that particular case. It does not come up often and I wonder if there are valid use cases for calling `.parse()` down tree in fancy custom programs.

```
const { program } = require('commander')
   .exitOverride()
   .option('-g, --global')
   .command('sub')
   .option('--sub-option')
   .parse(); // oops
```

To encourage a usage pattern that avoids several potential problems, I use this style as much as possible in the hopes people will just copy it:
- define `program`
- configure program
- new statement per subcommand
- call `.parse()`

```
import { program } from 'commander';

program
   .exitOverride()
   .option('-g, --global');

program.command('sub')
   .option('--sub-option');

program.parse();
```

Reply:
How about we propagate `parse()` / `parseAsync()` calls to the root command? Then chucking a call at the end of a subcommand definition would not be a problem, and we would not need the warning since subcommands would not be usable as stand-alones.

Reply:
The suggestion from the first comment would also become irrelevant.

Reply:
Hmmm... No, I think not. Either the user is intentionally doing something interesting outside normal usage, or they are confused about how Commander works. Doing something different that _might_ be what they intended is masking the situation.

Reply:
As of now, there are no meaningful semantics / intended usage patterns for `parse()` / `parseAsync()` calls on subcommands anyway, and any such call should be considered wrong. The only meaningful semantics for such a call I can think of are exactly that, propagating it to the root command. So why not assign a meaning to calls that currently have no meaning at all, giving the developer the freedom to chuck a parse call at the end of a subcommand chain? That is not _doing something different that might be what they intended_, that is doing the only meaningful thing when receiving such a call instead of doing complete nonsense (like what the library currently does) or having to come up with ways to tell the user they are using the library in a wrong way (like what this PR currently does).

Reply:
Two days later, I am having second thoughts about my comment on the suggestion from the first comment becoming irrelevant. I think I was wrong saying that, it will still be relevant because it deals with parse calls on a parent command that has lost ownership of a subcommand. So I would still like to hear why you disliked that suggestion. Implementing it would make the warning this thread is about unnecessary since parsing subcommands as stand-alones would become possible. Could still leave the warning for the case when parse call is chucked a the end of an implicit subcommand chain, though, for example by adding a `_parseCallsUndesired`  field with the default value of `false` to the `Command` class.